com.wamas.ide.launching.ui/src/com/wamas/ide/launching/ui/build/LcDslTargetPlatformSupport.java

```diff
@@ -82,444 +86,540 @@
-				.stream(PDECore.getDefault().getModelManager().getExternalModelManager().getAllModels()).parallel()
-				.flatMap(this::added).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, this::error,
-						this::createSortedUriMap));
-
-		lock.writeLock().lock();
-		try {
-			SortedMapDifference<URI, URI> difference = Maps.difference(uriMap, newContent);
-			if (!difference.entriesOnlyOnLeft().isEmpty()) {
-				Map<String, ToBeBuilt> partitioned = new HashMap<>();
-				difference.entriesOnlyOnLeft().keySet().forEach(uri -> {
-					pendingUpdated.remove(uri);
-					partitioned.computeIfAbsent(uri.segment(0), any -> new ToBeBuilt()).getToBeDeleted().add(uri);
-				});
-				partitioned.forEach(scheduler.get());
-			}
-
-			pendingUpdated.addAll(difference.entriesDiffering().keySet());
-			pendingUpdated.addAll(difference.entriesOnlyOnRight().keySet());
-			uriMap.clear();
-			uriMap.putAll(newContent);
-			if (!difference.areEqual()) {
-				touchAnyProject(Set.of());
-			}
-		} finally {
-			lock.writeLock().unlock();
-		}
-	}
-
-	private void touchAnyProject(Set<String> butNot) {
-		Job touchJob = Job.createSystem("Touch project after target platform change", new ICoreRunnable() {
-			@Override
-			public void run(IProgressMonitor monitor) throws CoreException {
-				IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
-				IProject candidate = null;
-				for (IProject p : root.getProjects()) {
-					if (XtextProjectHelper.hasBuilder(p)) {
-						if (butNot.contains(p.getName())) {
-							if (candidate == null) {
-								candidate = p;
-							}
-						} else {
-							p.touch(monitor);
-							return;
-						}
-					}
-				}
-				if (candidate != null) {
-					candidate.touch(monitor);
-					return;
-				}
-			}
-		});
-		touchJob.setRule(ResourcesPlugin.getWorkspace().getRoot());
-		touchJob.schedule(0);
-	}
-
-	@Override
-	public void reset() {
-		// nothing to do
-	}
-
-	@Override
-	public void reset(IProject project) {
-		// nothing to do
-	}
-
-	@Override
-	public boolean queueChange(Delta delta) {
-		// nothing to do
-		return false;
-	}
-
-	@Override
-	public boolean needsRebuild(IProject project, Collection<Delta> deltas) {
-		// nothing to do
-		return false;
-	}
-
-	@Override
-	public void createCheckpoint() {
-		lock.writeLock().lock();
-		try {
-			this.pendingUpdatedCheckpoint = new HashSet<>(pendingUpdated);
-			queuedBuildData.queueURIs(pendingUpdated);
-			this.pendingUpdated.clear();
-		} finally {
-			lock.writeLock().unlock();
-		}
-	}
-
-	@Override
-	public void discardCheckpoint() {
-		lock.writeLock().lock();
-		try {
-			this.pendingUpdatedCheckpoint = null;
-		} finally {
-			lock.writeLock().unlock();
-		}
-	}
-
-	@Override
-	public void rollback() {
-		lock.writeLock().lock();
-		try {
-			this.pendingUpdated = pendingUpdatedCheckpoint;
-			this.pendingUpdatedCheckpoint = null;
-		} finally {
-			lock.writeLock().unlock();
-		}
-	}
+public class LcDslTargetPlatformSupport
```

Comment:
It's unfortunate that the formatting settings of the previous revision did not match the ons from the projects. Sorry about that.

Reply:
All good :) It's not so strict :D